Xp login

anderklander opened the pull request:
Could not find any reasonable solution for tests of the footer visibility when just animating it out of sight, does anyone have any suggestions? 

Also noted that the cursor moves to the end after writing in a selected area, but it seems to be that way now as well so i didn't change anything.

Put a fix for not being able to use the placeholderTextColor, so this will change the placeholder text color on support as well, but now its according to design i zeplin!

Checklist for a PR:

* [ ] Describe the change in **`CHANGELOG.md`**. Only applicable if the change has any impact for a user.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/mullvad/mullvadvpn-app/79)
<!-- Reviewable:end -->

eriklarko commented:
*[app/components/Login.js, line 55 at r1](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79#-L6NkOglcZ7l8K23jT6U:-L6NkOgm7ZA-aP95Y5mi:byp7eq8) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/958adfe358fb3c28435597a3893887725a16536f/app/components/Login.js#L55)):*
> ```JavaScript
>       this.state.notifyOnFirstChangeAfterFailure = true;
>     }
>     this.state.dropdownAnimationStyle = Styles.createAnimatedViewStyle({height: this.state.animatedDropdownValue});
> ```

Very minor and also perhaps related to my style, so feel free to ignore. But maybe some line breaks here? Perhaps
```
Styles.createAnimatedViewStyle({
    height: this.state.animatedDropdownValue,
});
```

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79)*
<!-- Sent from Reviewable.io -->

eriklarko commented:
Reviewed 8 of 9 files at r1.
Review status: 8 of 9 files reviewed at latest revision, 2 unresolved discussions.

---

*[app/components/AccountInput.js, line 219 at r1](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79#-L6NiNfSHNqtklq_v63J:-L6NiNfSHNqtklq_v63K:b79kfcv) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/958adfe358fb3c28435597a3893887725a16536f/app/components/AccountInput.js#L219)):*
> ```JavaScript
>       e.preventDefault();
> 
>       //this._ignoreSelect = true;
> ```

Some debug stuff here and the console log later

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79)*
<!-- Sent from Reviewable.io -->

eriklarko commented:
Hmm, is it perhaps possible to access the `Animated` value on the footer and check stuff there?

---







*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79#-:-L6NlvKIMszYRhdACJHK:bf6wxef)*
<!-- Sent from Reviewable.io -->

anderklander commented:
Tried to do that, but got zero all the time ...

---

Review status: 7 of 9 files reviewed at latest revision, 3 unresolved discussions.

---

*[app/components/AccountInput.js, line 219 at r1](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79#-L6NiNfSHNqtklq_v63J:-L6QGK7qisDAjP-KSvvm:b-896fix) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/958adfe358fb3c28435597a3893887725a16536f/app/components/AccountInput.js#L219)):*
<details><summary><i>Previously, eriklarko (Erik Larkö) wrote…</i></summary><blockquote>

Some debug stuff here and the console log later
</blockquote></details>

Done.

---

*[app/components/Login.js, line 55 at r1](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79#-L6NkOglcZ7l8K23jT6U:-L6QGVKfXqIWdkoMZX2F:b-896fix) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/958adfe358fb3c28435597a3893887725a16536f/app/components/Login.js#L55)):*
<details><summary><i>Previously, eriklarko (Erik Larkö) wrote…</i></summary><blockquote>

Very minor and also perhaps related to my style, so feel free to ignore. But maybe some line breaks here? Perhaps
```
Styles.createAnimatedViewStyle({
    height: this.state.animatedDropdownValue,
});
```
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79#-:-L6QEEH1N8MTNHOkKNjK:b1p8e3s)*
<!-- Sent from Reviewable.io -->

eriklarko commented:
Ah, the animation probably doesn't run... It might be possible to add another test prop to the footer and check that value. It will not actually test the animation, but e.g. `<Animated.View testIsExpanded={some state}...>` and then check that prop in the test?

---

Reviewed 2 of 2 files at r2.
Review status: 8 of 9 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/79#-:-L6TACfL7YOjZzMm3uJx:b-z029o0)*
<!-- Sent from Reviewable.io -->

anderklander commented:
Added a test, might be a bit verbose in the tests, seems to work if we think its good enough.

---

Review status: 7 of 9 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79#-:-L6VF27vT0Ptdd82r5Dl:b2qoveu)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Reviewed 2 of 9 files at r1, 1 of 2 files at r2.
Review status: 7 of 9 files reviewed at latest revision, 1 unresolved discussion.

---

*[app/components/AccountInput.js, line 31 at r3](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79#-L6aAvqadW1uN3ij6MoL:-L6aAvqadW1uN3ij6MoM:b73mcx8) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/b35c95b35b260a6b65a323acc85fd64697f179e5/app/components/AccountInput.js#L31)):*
> ```JavaScript
> type SelectionRange = [number, number];
> 
> export default class AccountInput extends React.Component<AccountInputProps, AccountInputState> {
> ```

Maybe should inherit from `RX.Component`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Reviewed 1 of 9 files at r1.
Review status: 7 of 9 files reviewed at latest revision, 1 unresolved discussion.

---

*[app/components/Login.js, line 45 at r4](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/79#-L6aMC68D0wopfByte9e:-L6aMC68D0wopfByte9f:b-nt2cz4) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/6a1893f49f513eb2612117c6b15ef1b10bc71ebd/app/components/Login.js#L45)):*
> ```JavaScript
>     animatedFooterValue: Animated.createValue(0),
>     animatedDropdownValue: Animated.createValue(0),
>     animation: Animated.CompositeAnimation,
> ```

seems like you assign a class to the property that expects the instance of `CompositeAnimation`.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/79)*
<!-- Sent from Reviewable.io -->

anderklander commented:
Review status: 7 of 9 files reviewed at latest revision, 3 unresolved discussions.

---

*[app/components/AccountInput.js, line 31 at r3](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79#-L6aAvqadW1uN3ij6MoL:-L6aNG1EK6timk-yE4K4:b-896fix) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/b35c95b35b260a6b65a323acc85fd64697f179e5/app/components/AccountInput.js#L31)):*
<details><summary><i>Previously, pronebird (Andrei Mihailov) wrote…</i></summary><blockquote>

Maybe should inherit from `RX.Component`
</blockquote></details>

Done.

---

*[app/components/Login.js, line 45 at r4](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79#-L6aMC68D0wopfByte9e:-L6aNFFrTutdqcjonu3c:b-896fix) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/6a1893f49f513eb2612117c6b15ef1b10bc71ebd/app/components/Login.js#L45)):*
<details><summary><i>Previously, pronebird (Andrei Mihailov) wrote…</i></summary><blockquote>

seems like you assign a class to the property that expects the instance of `CompositeAnimation`.
</blockquote></details>

Done.

---

*[app/components/Login.js, line 46 at r4](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79#-L6aMLGcASt0u2hNF7ai:-L6aNETY4wZX1ol5KoYB:b-896fix) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/6a1893f49f513eb2612117c6b15ef1b10bc71ebd/app/components/Login.js#L46)):*
<details><summary><i>Previously, pronebird (Andrei Mihailov) wrote…</i></summary><blockquote>

Same here, you probably want to assign an actual style or null
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79)*
<!-- Sent from Reviewable.io -->

pronebird commented:
That's how the app looks on startup:

![Screenshot 2018-03-02 12.04.42.png](https://files.reviewableusercontent.io/images/reviewable/704044/nFeEGyLAsXmK1vSwZK3_9mLA/Screenshot%202018-03-02%2012.04.42.png)

---

Reviewed 1 of 1 files at r4.
Review status: 7 of 9 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/79#-:-L6aNjDPDaxhn1kF5EVJ:b-jcrn79)*
<!-- Sent from Reviewable.io -->

anderklander commented:
onFocus was not called by autoFocus, so changed the initial state to isActive=true

---

Review status: 7 of 9 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79#-:-L6aQePy0zNCj9xbp6NW:bxfb3sp)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Review status: 7 of 9 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[test/helpers/dom-events.js, line 21 at r5](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79#-L6a_Hn5ZwUNxnnsMstU:-L6a_Hn5ZwUNxnnsMstV:bb76mu4) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/94926944ff71ad7b28d144ad2614959783b42f6c/test/helpers/dom-events.js#L21)):*
> ```JavaScript
> 
> export function createKeyEvent(key: Keycode): Object {
>   return Object.assign({}, { key }, keycodes[key], {preventDefault: () => {}});
> ```

Why?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Reviewed 1 of 2 files at r3.
Review status: 8 of 9 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Review status: 8 of 9 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[test/components/AccountInput.spec.js, line 144 at r5](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79#-L6aaPRMgOYnFvNijL4Z:-L6aaPRMgOYnFvNijL4_:bl9rkha) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/94926944ff71ad7b28d144ad2614959783b42f6c/test/components/AccountInput.spec.js#L144)):*
> ```JavaScript
> 
>     component.setState({}, () => {
>       expect(component.state().value).to.be.equal('111111111111');
> ```

Is it shallow renderer that turns `state` into function?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Reviewed 1 of 1 files at r5.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[test/helpers/dom-events.js, line 21 at r5](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79#-L6a_Hn5ZwUNxnnsMstU:-L6abBLpw1bFK3v9HXhr:bjnh9x5) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/94926944ff71ad7b28d144ad2614959783b42f6c/test/helpers/dom-events.js#L21)):*
<details><summary><i>Previously, pronebird (Andrei Mihailov) wrote…</i></summary><blockquote>

Why?
</blockquote></details>

Alright I realized that event simulation is crooked in shallow renderer. I am a little concerned about mocking the text input that way since we actually block the backspace etc.. but I guess there is no way around that. I'd love a comment left here to reflect why exactly preventDefault is mocked in such way.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Reviewed 1 of 9 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Reviewed 1 of 9 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Reviewed 1 of 9 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[app/components/Login.js, line 32 at r5](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79#-L6aepFCe9BnDy1Y4STD:-L6aepFCe9BnDy1Y4STE:b-1dyp7r) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/94926944ff71ad7b28d144ad2614959783b42f6c/app/components/Login.js#L32)):*
> ```JavaScript
>   animatedFooterValue: Animated.Value,
>   animatedDropdownValue: Animated.Value,
>   animation: Animated.CompositeAnimation,
> ```

Can be null. Not sure why flow does not pick this up.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79)*
<!-- Sent from Reviewable.io -->

anderklander commented:
Review status: 7 of 9 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[test/components/AccountInput.spec.js, line 144 at r5](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79#-L6aaPRMgOYnFvNijL4Z:-L6agFRYM6kechqbF__Z:bb77ufg) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/94926944ff71ad7b28d144ad2614959783b42f6c/test/components/AccountInput.spec.js#L144)):*
<details><summary><i>Previously, pronebird (Andrei Mihailov) wrote…</i></summary><blockquote>

Is it shallow renderer that turns `state` into function?
</blockquote></details>

Yes.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Could you please bring back the animations we had on account dropdown items and green arrow?

They looked nice. :)

---

Review status: 7 of 9 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79#-:-L6aiFxCfrz8GOl7_vK6:b-11f6si)*
<!-- Sent from Reviewable.io -->

pronebird commented:
I tested the component a little. It seems that we have an error if animation happens when I click on Settings button in the top right corner.

![Screenshot 2018-03-02 13.41.01.png](https://files.reviewableusercontent.io/images/reviewable/704044/ae_9s5gIyO1WunXq6CvUo4Nd/Screenshot%202018-03-02%2013.41.01.png)

---

Review status: 7 of 9 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79#-:-L6aikW1IIfPNEqIsqxr:b-judgji)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Reviewed 2 of 2 files at r6.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[app/components/AccountInput.js, line 211 at r6](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79#-L6bQaNyuoVUbE-eTpWO:-L6bQaNyuoVUbE-eTpWP:b-frtxoi) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/2212e82dbf2b8a48c895c2e3d3a5485a03ba032d/app/components/AccountInput.js#L211)):*
> ```JavaScript
> 
>   // Events
>   _ignoreSelect: boolean;
> ```

I really had to document why I do that. My best bet is that `onSelect` is dispatched when we programmatically change the input.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[app/components/AccountInput.js, line 211 at r6](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79#-L6bQaNyuoVUbE-eTpWO:-L6bRe0_Q0BmPIp-dQUv:b-s55t0x) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/2212e82dbf2b8a48c895c2e3d3a5485a03ba032d/app/components/AccountInput.js#L211)):*
<details><summary><i>Previously, pronebird (Andrei Mihailov) wrote…</i></summary><blockquote>

I really had to document why I do that. My best bet is that `onSelect` is dispatched when we programmatically change the input.
</blockquote></details>

If you're skiing next week I'll probably document that then. :skier: 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79)*
<!-- Sent from Reviewable.io -->

jvff commented:
Reviewed 4 of 9 files at r1, 1 of 2 files at r3, 2 of 2 files at r6, 4 of 4 files at r7.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/79)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Reviewed 2 of 4 files at r7.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[package.json, line 30 at r7](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/79#-L87-lvG1FtulWTEqkWp:-L87-lvG1FtulWTEqkWq:b-x3udpu) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/84d330807ddc30e1e618e1cd7f63e9981fcb5ff7/package.json#L30)):*
> ```JSON
>     "react-router-redux": "^5.0.0-alpha.9",
>     "react-simple-maps": "^0.10.1",
>     "reactxp": "^1.0.2",
> ```

```
warning " > reactxp@1.0.2" has incorrect peer dependency "react-native@^0.53.3".
warning " > reactxp@1.0.2" has incorrect peer dependency "react-native-windows@^0.51.0-rc.1".
```

Please correct the peer dependencies when updating the dependent modules.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/79)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Reviewed 1 of 4 files at r7.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Reviewed 1 of 4 files at r7.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Review status: all files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[app/components/Login.js, line 384 at r7](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79#-L878CVkLNJjLCU5iwy7:-L878CVkLNJjLCU5iwy8:boc0ljs) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/84d330807ddc30e1e618e1cd7f63e9981fcb5ff7/app/components/Login.js#L384)):*
> ```JavaScript
> };
> 
> class AccountDropdown extends React.Component<AccountDropdownProps> {
> ```

The account dropdown does not resize after removing the item from the list anymore.

![Screenshot 2018-03-21 11.39.07.png](https://files.reviewableusercontent.io/images/reviewable/704044/O4-DBZ4u7dYz5NKFXX934Yy_/Screenshot%202018-03-21%2011.39.07.png)

Could you please do more vigorous testing?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---

*[app/components/Login.js, line 408 at r7](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79#-L878yTUUFu4bGQiuat_:-L878yTUUFu4bGQiuata:b-nv6cg2) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/84d330807ddc30e1e618e1cd7f63e9981fcb5ff7/app/components/Login.js#L408)):*
> ```JavaScript
> };
> 
> class AccountDropdownItem extends React.Component<AccountDropdownItemProps> {
> ```

Dropdown items used to have a slight hover. same for buttons inside.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79)*
<!-- Sent from Reviewable.io -->

anderklander commented:
Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---

*[package.json, line 30 at r7](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/79#-L87-lvG1FtulWTEqkWp:-L87CC0z3UjOjFU_Czve:b-s5xf0h) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/84d330807ddc30e1e618e1cd7f63e9981fcb5ff7/package.json#L30)):*
<details><summary><i>Previously, pronebird (Andrei Mihailov) wrote…</i></summary><blockquote>

```
warning " > reactxp@1.0.2" has incorrect peer dependency "react-native@^0.53.3".
warning " > reactxp@1.0.2" has incorrect peer dependency "react-native-windows@^0.51.0-rc.1".
```

Please correct the peer dependencies when updating the dependent modules.

</blockquote></details>

The best thing would be if we could remove react-native-windows, its big and useless (from what i can tell?) for us ... Dont know if Microsoft could accept to change it to an optional, i tried to remove it but electron builder throws errors, have no experience in electronbuilder, could it be possible to configure this in some way?

And if we still have to keep R-N-W it will still throw errors because xp has peer deps that have different peer dep-version so there will still be one warning.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/79)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---

*[package.json, line 30 at r7](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79#-L87-lvG1FtulWTEqkWp:-L87Gjrjjsl5H3GwtvbQ:bch9san) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/84d330807ddc30e1e618e1cd7f63e9981fcb5ff7/package.json#L30)):*
<details><summary><i>Previously, anderklander wrote…</i></summary><blockquote>

The best thing would be if we could remove react-native-windows, its big and useless (from what i can tell?) for us ... Dont know if Microsoft could accept to change it to an optional, i tried to remove it but electron builder throws errors, have no experience in electronbuilder, could it be possible to configure this in some way?

And if we still have to keep R-N-W it will still throw errors because xp has peer deps that have different peer dep-version so there will still be one warning.
</blockquote></details>

Not sure what they use it for but it's listed in peer dependencies which means RX needs this for something.

You can raise an issue with them if you wish but in the immediate short term it would be easier to line up with the requirements and use the same versions:

```
 "react-native": "^0.53.3",
 "react-native-windows": "^0.51.0-rc.1"
```

See https://github.com/Microsoft/reactxp/blob/master/package.json#L26

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79)*
<!-- Sent from Reviewable.io -->

anderklander commented:
Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---

*[package.json, line 30 at r7](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/79#-L87-lvG1FtulWTEqkWp:-L87Hu8rCchB122Ju88T:bgcla0i) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/84d330807ddc30e1e618e1cd7f63e9981fcb5ff7/package.json#L30)):*
<details><summary><i>Previously, pronebird (Andrei Mihailov) wrote…</i></summary><blockquote>

Not sure what they use it for but it's listed in peer dependencies which means RX needs this for something.

You can raise an issue with them if you wish but in the immediate short term it would be easier to line up with the requirements and use the same versions:

```
 "react-native": "^0.53.3",
 "react-native-windows": "^0.51.0-rc.1"
```

See https://github.com/Microsoft/reactxp/blob/master/package.json#L26
</blockquote></details>

As i wrote before, we will still have warnings:

warning "react-native-windows@0.51.0-rc.4" has incorrect peer dependency "react@16.0.0".
warning "react-native-windows@0.51.0-rc.4" has incorrect peer dependency "react-native@~0.51.0".

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/79)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---

*[package.json, line 30 at r7](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79#-L87-lvG1FtulWTEqkWp:-L87J82YhmWoBHSNcWXu:btrhwwp) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/84d330807ddc30e1e618e1cd7f63e9981fcb5ff7/package.json#L30)):*
<details><summary><i>Previously, anderklander wrote…</i></summary><blockquote>

As i wrote before, we will still have warnings:

warning "react-native-windows@0.51.0-rc.4" has incorrect peer dependency "react@16.0.0".
warning "react-native-windows@0.51.0-rc.4" has incorrect peer dependency "react-native@~0.51.0".
</blockquote></details>

Yeah you're right. I just raised an issue with RX to see what's their take on that:

https://github.com/Microsoft/reactxp/issues/562

I still think if we at least line up with RX then react-native-windows will catch up in the coming RC.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79)*
<!-- Sent from Reviewable.io -->

anderklander commented:
Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---

*[package.json, line 30 at r7](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/79#-L87-lvG1FtulWTEqkWp:-L87JZLa7OKKqR6lsQZM:b-rne9mx) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/84d330807ddc30e1e618e1cd7f63e9981fcb5ff7/package.json#L30)):*
<details><summary><i>Previously, pronebird (Andrei Mihailov) wrote…</i></summary><blockquote>

Yeah you're right. I just raised an issue with RX to see what's their take on that:

https://github.com/Microsoft/reactxp/issues/562

I still think if we at least line up with RX then react-native-windows will catch up in the coming RC.
</blockquote></details>

I still think it would be great if we could get rid of R-N-W since the module folder is almost 50MB, and i guess that reactxp does not need it unless building UWP-apps.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/79)*
<!-- Sent from Reviewable.io -->

anderklander commented:
Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---

*[app/components/Login.js, line 408 at r7](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/79#-L878yTUUFu4bGQiuat_:-L87M0zL5HNYzkfDNua8:b-oapugv) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/84d330807ddc30e1e618e1cd7f63e9981fcb5ff7/app/components/Login.js#L408)):*
<details><summary><i>Previously, pronebird (Andrei Mihailov) wrote…</i></summary><blockquote>

Dropdown items used to have a slight hover. same for buttons inside.
</blockquote></details>

Will change so we use the CellButton-component and get the hover-logic needed there.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/79)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---

*[package.json, line 30 at r7](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79#-L87-lvG1FtulWTEqkWp:-L87MR7i3eqCRHMu6n-g:bcay812) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/84d330807ddc30e1e618e1cd7f63e9981fcb5ff7/package.json#L30)):*
<details><summary><i>Previously, anderklander wrote…</i></summary><blockquote>

I still think it would be great if we could get rid of R-N-W since the module folder is almost 50MB, and i guess that reactxp does not need it unless building UWP-apps.
</blockquote></details>

Oh I see right, we don't actually need react-native-windows for desktop builds.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks failed.

---

*[app/components/Login.js, line 408 at r7](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79#-L878yTUUFu4bGQiuat_:-L87N_kd9lmRUR84toIL:b6x532s) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/84d330807ddc30e1e618e1cd7f63e9981fcb5ff7/app/components/Login.js#L408)):*
<details><summary><i>Previously, anderklander wrote…</i></summary><blockquote>

Will change so we use the CellButton-component and get the hover-logic needed there.
</blockquote></details>

Thanks :+1: 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Review status: all files reviewed at latest revision, 5 unresolved discussions, some commit checks failed.

---

*[app/components/Login.js, line 356 at r7](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79#-L87NkR2fLXR1L_ybQ3G:-L87NkR2fLXR1L_ybQ3H:b-cl1eoj) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/84d330807ddc30e1e618e1cd7f63e9981fcb5ff7/app/components/Login.js#L356)):*
> ```JavaScript
>         </View>
>         <Animated.View style={ this.state.dropdownAnimationStyle }>
>           <View onLayout={this._onDropdownLayout} ref={ this._onAccountDropdownContainerRef }>
> ```

There is other PR that I've sent today just to point out that we could use Accordion.

This won't work on Android for now but if you develop on desktop then it should work and we can manage the Accordion conversion to RX in a separate PR.

Less code in Login, less repetition and less issues to tackle right now.

Take a look at https://github.com/mullvad/mullvadvpn-app/pull/102

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Review status: all files reviewed at latest revision, 5 unresolved discussions, some commit checks failed.

---

*[package.json, line 30 at r7](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79#-L87-lvG1FtulWTEqkWp:-L87PRSi-BJ85M0EMQu0:b-pe3bgf) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/84d330807ddc30e1e618e1cd7f63e9981fcb5ff7/package.json#L30)):*
<details><summary><i>Previously, pronebird (Andrei Mihailov) wrote…</i></summary><blockquote>

Oh I see right, we don't actually need react-native-windows for desktop builds.
</blockquote></details>

Regardless issues with `react-native-windows`, `react-native`  still has wrong peer dependency. you cannot upgrade `reactxp` without checking that peer dependencies match. This may cause all kind of hard to track issues such as what happened during the switch from React 15 -> 16.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79)*
<!-- Sent from Reviewable.io -->

anderklander commented:
Review status: all files reviewed at latest revision, 5 unresolved discussions, some commit checks failed.

---

*[package.json, line 30 at r7](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/79#-L87-lvG1FtulWTEqkWp:-L87RV2HCD508mplRvks:bkn23g8) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/84d330807ddc30e1e618e1cd7f63e9981fcb5ff7/package.json#L30)):*
<details><summary><i>Previously, pronebird (Andrei Mihailov) wrote…</i></summary><blockquote>

Regardless issues with `react-native-windows`, `react-native`  still has wrong peer dependency. you cannot upgrade `reactxp` without checking that peer dependencies match. This may cause all kind of hard to track issues such as what happened during the switch from React 15 -> 16.
</blockquote></details>

Since the project in its current state does not build on react-native i didnt bother to change that version, as changing it would not remove or decrease the number of warnings. But when we have something that builds on android i totally agree. But i will update in this PR as well.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/79)*
<!-- Sent from Reviewable.io -->

anderklander commented:
Review status: 7 of 13 files reviewed at latest revision, 5 unresolved discussions, some commit checks failed.

---

*[app/components/Login.js, line 356 at r7](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/79#-L87NkR2fLXR1L_ybQ3G:-L88GlK5DzmT1s8ma3uE:b-896fix) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/84d330807ddc30e1e618e1cd7f63e9981fcb5ff7/app/components/Login.js#L356)):*
<details><summary><i>Previously, pronebird (Andrei Mihailov) wrote…</i></summary><blockquote>

There is other PR that I've sent today just to point out that we could use Accordion.

This won't work on Android for now but if you develop on desktop then it should work and we can manage the Accordion conversion to RX in a separate PR.

Less code in Login, less repetition and less issues to tackle right now.

Take a look at https://github.com/mullvad/mullvadvpn-app/pull/102
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/79)*
<!-- Sent from Reviewable.io -->

anderklander commented:
In order to have separate hover-look on the close icon in the dropdown i guess we need to modify the Img-component or add some CSS-styling of hovering, i guess its ok if we leave mobile OS with mouse for the time beeing.

Spent some time on it but could not come up with a good solution.

---

Review status: 7 of 13 files reviewed at latest revision, 5 unresolved discussions.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/79#-:-L88IIjy6RxLzyjOhQ9f:b-hfr0ui)*
<!-- Sent from Reviewable.io -->

jvff commented:
Reviewed 6 of 6 files at r8.
Review status: all files reviewed at latest revision, 5 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/79)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Review status: 10 of 14 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[app/components/Login.js, line 384 at r7](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79#-L878CVkLNJjLCU5iwy7:-L8WpEcUKvLN4VCZ_lhT:b-14hi8j) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/84d330807ddc30e1e618e1cd7f63e9981fcb5ff7/app/components/Login.js#L384)):*
<details><summary><i>Previously, pronebird (Andrei Mihailov) wrote…</i></summary><blockquote>

The account dropdown does not resize after removing the item from the list anymore.

![Screenshot 2018-03-21 11.39.07.png](https://files.reviewableusercontent.io/images/reviewable/704044/O4-DBZ4u7dYz5NKFXX934Yy_/Screenshot%202018-03-21%2011.39.07.png)

Could you please do more vigorous testing?
</blockquote></details>

Accordion handles this properly

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Reviewed 1 of 4 files at r9.
Review status: 11 of 14 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Review status: 11 of 14 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[app/components/Login.js, line 295 at r9](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79#-L8WqMWtCYu30n41rTK_:-L8WqMWtCYu30n41rTKa:b-amco8h) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/44c0b3e8fe2d2af5c04346df9908e03c11bd3323/app/components/Login.js#L295)):*
> ```JavaScript
> 
>   // container element used for measuring the height of the accounts dropdown
>   _accountDropdownElement: ?React.Node;
> ```

Is it used anywhere now?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Review status: 11 of 14 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[app/components/Login.js, line 296 at r9](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79#-L8WqP7rZGoumpXqD1ZQ:-L8WqP7rZGoumpXqD1ZR:buo9nmt) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/44c0b3e8fe2d2af5c04346df9908e03c11bd3323/app/components/Login.js#L296)):*
> ```JavaScript
>   // container element used for measuring the height of the accounts dropdown
>   _accountDropdownElement: ?React.Node;
>   _onAccountDropdownContainerRef = ref => this._accountDropdownElement = ref;
> ```

Same here

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79)*
<!-- Sent from Reviewable.io -->

anderklander commented:
Review status: 10 of 14 files reviewed at latest revision, 3 unresolved discussions.

---

*[app/components/Login.js, line 295 at r9](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/79#-L8WqMWtCYu30n41rTK_:-L8WrFoI8Ksbm0RtEIMY:bga88j7) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/44c0b3e8fe2d2af5c04346df9908e03c11bd3323/app/components/Login.js#L295)):*
<details><summary><i>Previously, pronebird (Andrei Mihailov) wrote…</i></summary><blockquote>

Is it used anywhere now?
</blockquote></details>

No! Removed.

---

*[app/components/Login.js, line 296 at r9](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/79#-L8WqP7rZGoumpXqD1ZQ:-L8WrIqp3hRaAQs2Do77:b-v6s1op) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/44c0b3e8fe2d2af5c04346df9908e03c11bd3323/app/components/Login.js#L296)):*
<details><summary><i>Previously, pronebird (Andrei Mihailov) wrote…</i></summary><blockquote>

Same here
</blockquote></details>

Removed.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/79)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Reviewed 2 of 2 files at r10.
Review status: 12 of 14 files reviewed at latest revision, 2 unresolved discussions.

---

*[app/components/LoginStyles.js, line 86 at r10](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79#-L8WsTHxGGvbY_wMzUfb:-L8WsTHxGGvbY_wMzUfc:bg75qia) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/0d08eb3f80ae8db32be3a9e842c104ca8898ae52/app/components/LoginStyles.js#L86)):*
> ```JavaScript
>       paddingTop: 10,
>       paddingRight: 12,
>       paddingLeft: 12,
> ```

Can we align the leading space so it matches the start of the text in input field. Currently the text in dropdown is like being pushed too much inside

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Review status: 12 of 14 files reviewed at latest revision, 3 unresolved discussions.

---

*[app/components/LoginStyles.js, line 83 at r10](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79#-L8WtAK3bxNSk8MSX6ms:-L8WtAK3bxNSk8MSX6mt:b-mz4n57) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/0d08eb3f80ae8db32be3a9e842c104ca8898ae52/app/components/LoginStyles.js#L83)):*
> ```JavaScript
>       backgroundColor: colors.darkBlue,
>     },
>     account_dropdown__item: {
> ```

It's usually easier in CSS. The last item in the dropdown has `margin-bottom: 1px` which adds a visible space between the outer border and the dropdown item itself. Would be awesome to remove it for the last item in dropdown.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79)*
<!-- Sent from Reviewable.io -->

anderklander commented:
Review status: 11 of 14 files reviewed at latest revision, 3 unresolved discussions.

---

*[app/components/LoginStyles.js, line 83 at r10](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/79#-L8WtAK3bxNSk8MSX6ms:-L8WvBkv2p3G54VnzxOH:b-896fix) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/0d08eb3f80ae8db32be3a9e842c104ca8898ae52/app/components/LoginStyles.js#L83)):*
<details><summary><i>Previously, pronebird (Andrei Mihailov) wrote…</i></summary><blockquote>

It's usually easier in CSS. The last item in the dropdown has `margin-bottom: 1px` which adds a visible space between the outer border and the dropdown item itself. Would be awesome to remove it for the last item in dropdown.
</blockquote></details>

Done.

---

*[app/components/LoginStyles.js, line 86 at r10](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/79#-L8WsTHxGGvbY_wMzUfb:-L8WvC3s2_tIyaTV5RA0:b-896fix) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/0d08eb3f80ae8db32be3a9e842c104ca8898ae52/app/components/LoginStyles.js#L86)):*
<details><summary><i>Previously, pronebird (Andrei Mihailov) wrote…</i></summary><blockquote>

Can we align the leading space so it matches the start of the text in input field. Currently the text in dropdown is like being pushed too much inside
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/79)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Review status: 11 of 14 files reviewed at latest revision, 2 unresolved discussions.

---

*[app/components/styled/CellButton.js, line 118 at r10](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79#-L8WvhySY0ltacFRrtI1:-L8WvhySY0ltacFRrtI2:bcwwo79) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/0d08eb3f80ae8db32be3a9e842c104ca8898ae52/app/components/styled/CellButton.js#L118)):*
> ```JavaScript
> 
>               return React.cloneElement(node, updatedProps);
>             } else if (node != null){
> ```

In strict mode it's better to use `!==` or `if(node)`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Review status: 11 of 14 files reviewed at latest revision, 3 unresolved discussions.

---

*[app/components/AccountInput.js, line 84 at r10](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79#-L8WwKN3ZIgAfuloUb8U:-L8WwKN3ZIgAfuloUb8V:bkpjbak) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/0d08eb3f80ae8db32be3a9e842c104ca8898ae52/app/components/AccountInput.js#L84)):*
> ```JavaScript
>         onSelectionChange={ this.onSelect }
>         onPaste={ this.onPaste }
>         onCut={ this.onCut }
> ```

Funny but it seems that :fearful: there is no `onCut` on TextInput.. Cut functionality is broken now...

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Review status: 11 of 14 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[app/components/AccountInput.js, line 84 at r10](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79#-L8WwKN3ZIgAfuloUb8U:-L8WxKqZV5gZnGCyjGd-:bdf4xzb) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/0d08eb3f80ae8db32be3a9e842c104ca8898ae52/app/components/AccountInput.js#L84)):*
<details><summary><i>Previously, pronebird (Andrei Mihailov) wrote…</i></summary><blockquote>

Funny but it seems that :fearful: there is no `onCut` on TextInput.. Cut functionality is broken now...
</blockquote></details>

Raised an issue here https://github.com/Microsoft/reactxp/issues/574

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Review status: 11 of 14 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[app/components/AccountInput.js, line 211 at r6](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79#-L6bQaNyuoVUbE-eTpWO:-L8X2YkwoC_l3rgyFrfA:bbbox7t) ([raw file](https://github.com/mullvad/mullvadvpn-app/blob/2212e82dbf2b8a48c895c2e3d3a5485a03ba032d/app/components/AccountInput.js#L211)):*
<details><summary><i>Previously, pronebird (Andrei Mihailov) wrote…</i></summary><blockquote>

If you're skiing next week I'll probably document that then. :skier: 
</blockquote></details>

I hardly can remember why I used this and now the value of this flag is even more distorted. I'll leave it as is until the next revision since we have plenty of other things to do.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Reviewed 1 of 6 files at r8.
Review status: 11 of 14 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Reviewed 2 of 6 files at r8.
Review status: 11 of 14 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Reviewed 1 of 4 files at r9.
Review status: 11 of 14 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Reviewed 1 of 4 files at r9.
Review status: 12 of 14 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Known regression: AccountInput won't support ONCUT until https://github.com/Microsoft/reactxp/pull/575 is merged and we update the RX dependency.

---

Review status: 12 of 14 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79#-:-L8X3f0MmM-Bl-zcg-5t:bnkw8vz)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Phew a lot of work you put in this PR. :sweat: The last comment left and I think we should merge this.

---

Review status: 12 of 14 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mullvad/mullvadvpn-app/79#-:-L8X4BWPOFb6r1284EBy:b70ltps)*
<!-- Sent from Reviewable.io -->

anderklander commented:
Woho! Finally!

---

Review status: 11 of 14 files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/79#-:-L8X88NBBVcEav9GuDZ4:bz6ffsx)*
<!-- Sent from Reviewable.io -->

pronebird commented:
Reviewed 3 of 3 files at r11.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://beta.reviewable.io/reviews/mullvad/mullvadvpn-app/79)*
<!-- Sent from Reviewable.io -->
